File: vdp-ruby/test/visacheckout/get_payment_data_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class GetPaymentDataTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @visa_api_client = VisaAPIClient.new
  end
  
  def test_getPaymentInfo
    base_uri = 'wallet-services-web/'
    api_key = @config['apiKey']
    resource_path = 'payment/data/{callId}'
    resource_path = resource_path.sub('{callId}',@config['checkoutCallId'])
    response_code = @visa_api_client.doXPayTokenRequest(base_uri, resource_path, "apikey=#{api_key}", "Get Payment Information Test", "get", '')
    assert_equal("200", response_code, "Get Payment Information test failed")
    end
end

File: visa-dev-samplecode-master/vdp-node/test/visacheckout/getpaymentdatatest.js
var request = require('request');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var config = require('../../config/configuration.json');
var assert = require('chai').assert;
var req = request.defaults();
var randomstring = require('randomstring');

describe('Get Payment Data', function() {
	var visaAPIClient = new VisaAPIClient();

	it('Get Payment Information Test',function(done){
		this.timeout(10000);
		var apiKey = config.apiKey;
		var baseUri = 'wallet-services-web/';
		var resourcePath = 'payment/data/{callId}';
		resourcePath = resourcePath.replace('{callId}', config.checkoutCallId);
		var queryParams = 'apikey=' + apiKey;
		visaAPIClient.doXPayRequest(baseUri, resourcePath, queryParams, '', 'GET', {}, 
		function(err, responseCode) {
			if(!err) {
				assert.equal(responseCode, 200);
			} else {
				assert(false);
			}
			done();
		});
	});
});
